File: src/test/java/discretemath/structure/function/concrete/CompoundInterestTest.java
package discretemath.structure.function.concrete;

import java.math.BigDecimal;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class CompoundInterestTest {

	@Test
	void calculate() {
		CompoundInterest compoundInterest = new CompoundInterest(BigDecimal.valueOf(0.05), 12, 10);

		assertEquals(BigDecimal.valueOf(8235.0474885), compoundInterest.calculate(BigDecimal.valueOf(5000)));
	}
}

File: src/test/scala/discretemath/test/structure/function/concrete/CompoundInterestTest.scala
package discretemath.test.structure.function.concrete

import discretemath.structure.function.concrete.CompoundInterest
import org.scalatest.funsuite.AnyFunSuite

import java.math.BigDecimal;

class CompoundInterestTest extends AnyFunSuite {

  test("calculate") {
    val compoundInterest = new CompoundInterest(BigDecimal.valueOf(0.05), 12, 10);

    assert(compoundInterest.calculate(BigDecimal.valueOf(5000)) == 8235.0474885);
  }
}